include/libcyphal/transport/can/can_transport.hpp

```diff
@@ -0,0 +1,141 @@
+    ///
+    /// If set, this handler is called by the transport layer when a transient media related error occurs during
+    /// transport's (or any of its sessions) `run` method. A TX session `send` method may also trigger this handler.
+    ///
+    /// Note that there is a limited set of things that can be done within this handler, f.e.:
+    /// - it's not allowed to call transport's (or its session's) `run` method from within this handler;
+    /// - it's not allowed to call a TX session `send` or RX session `receive` methods from within this handler;
+    /// - main purpose of the handler:
+    ///   - is to log/report/stat the error;
+    ///   - potentially modify state of some "culprit" media related component (f.e. reset HW CAN controller);
+    ///   - return an optional (maybe different) error back to the transport.
+    /// - result error from the handler affects:
+    ///   - whether or not other redundant media of this transport will continue to be processed
+    ///     as part of this current "problematic" run (see description of return below),
+    ///   - propagation of the error up to the original user's call (result of the `run` or `send` methods).
+    ///
+    /// @param report The error report to be handled. It's made as non-const ref to allow the handler modify it,
+    ///               and f.e. reuse original `.error` field value by moving it as is to return result.
+    /// @return An optional (maybe different) error back to the transport.
+    ///         - If `cetl::nullopt` is returned, the original error (in the `report`) is considered as handled
+    ///           and insignificant for the transport. Transport will continue its current process (effectively
+    ///           either ignoring such transient failure, or retrying the process later on its next run).
+    ///         - If an error is returned, the transport will immediately stop current process, won't process any
+    ///           other media (if any), and propagate the returned error to the user (as result of `run` or etc).
+    ///
+    using TransientErrorHandler =
+        cetl::pmr::function<cetl::optional<AnyError>(TransientErrorReport::Variant& report_var), sizeof(void*) * 3>;
+
+    ICanTransport(const ICanTransport&)                = delete;
+    ICanTransport(ICanTransport&&) noexcept            = delete;
+    ICanTransport& operator=(const ICanTransport&)     = delete;
+    ICanTransport& operator=(ICanTransport&&) noexcept = delete;
+
+    /// Sets new transient error handler.
+    ///
+    /// If the handler is set, it will be called by the transport layer when a transient media related error occurs.
+    /// If the handler is not set (default mode), the transport will ignore such errors and continue
+    /// its current process in a "best-effort" manner, namely in assumption that
+    /// either other redundant media (if any) will deliver what is needed,
+    /// or later retry (aka next `run`) of the operation will resolve the issue.
```

Comment:
I don't think it's sustainable to ignore errors by default. If the client neglected or forgot to install the handler, the default behavior should be to escalate each error such that it breaks the sending/running/receiving loop. May I recommend we define a default transient error handler and set it in the constructor, such that there is always some handler available, allowing the client to override it if necessary? Having the handler always available could also slightly simplify the internals.

---

I looked at the actual implementation after writing the above comment and noticed that it appears to be implemented correctly: if the handler is not installed, the errors break the loop. Am I misreading the code?

Reply:
Yes, I also believe that implementation is correct. I need rewrite this comment to reflect actual implementation.